File: Modules/SEDO/EMS/SIE.Web.EMS/FixedAssets/Accounts/Commands/AddFixedAssetAccountCommand.cs
using SIE.EMS.FixedAssets.Accounts;
using SIE.Web.Command;

namespace SIE.Web.EMS.FixedAssets.Accounts.Commands
{
    /// <summary>
    /// 添加
    /// </summary>
    [JsCommand("SIE.Web.EMS.FixedAssets.Accounts.Commands.AddFixedAssetAccountCommand")]
    public class AddFixedAssetAccountCommand:ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var data = args.Data.ToJsonObject<FixedAssetsAccount>();
            data.Code = RT.Service.Resolve<FixedAssetsAccountController>().GetCode();
            return data;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS/FixedAssets/Accounts/Commands/AddFixedAssetAccountCommand.js
SIE.defineCommand('SIE.Web.EMS.FixedAssets.Accounts.Commands.AddFixedAssetAccountCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },
    showView: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: model,
                success: function (res) {
                    var data = res.Result;
                    CRT.Workbench.addPage({
                        entityType: me.view.model,
                        recordId: entity.getId(),
                        title: me.getEditViewTitle(entity),
                        isNew: true,
                        params: {
                            Code: data.Code,
                            token: me.view.token,
                            woTabId: CRT.Workbench.getTabPanel().getActiveTab().getId(),
                            action: 0
                        },
                        isDetail: true
                    });
                }
            }, me.view);
        }
    }
});